runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -467,28 +438,26 @@ public void addTrimSnapshot(@NonNull long trimMark, @NonNull long timestamp) {
     }
 
     /**
-     * Register SystemDownHandler.
-     * Please use CorfuRuntimeParameters builder to register this.
+     * These two handlers are provided to give some control on what happen when system is down.
+     *
+     * For applications that want to have specific behaviour when a the system appears unavailable, they can
+     * register their own handler for both before the rpc request and upon network exception.
+     *
+     * An example of how to use these handlers implementing timeout is given in
+     * test/src/test/java/org/corfudb/runtime/CorfuRuntimeTest.java
      *
-     * @param handler Handler to invoke in case of system Down.
-     * @return CorfuRuntime instance.
      */
-    @Deprecated
+    public Runnable beforeRpcHandler = () -> {};
+    public Runnable systemDownHandler = () -> {};
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make systemDownHandler a static final constant or non-public and provide accessors if needed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)